ledger/trie/trie_test.go

```diff
@@ -0,0 +1,276 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package trie_test
+
+import (
+	"encoding/hex"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/utils"
+	reference "github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+
+	"github.com/optakt/flow-dps/ledger/store"
+	"github.com/optakt/flow-dps/ledger/trie"
+	"github.com/optakt/flow-dps/testing/helpers"
+	"github.com/optakt/flow-dps/testing/mocks"
+)
+
+func TestTrie_InsertLeftRegister(t *testing.T) {
+	store := mocks.BaselineStore()
+	trie := trie.NewEmptyTrie(store)
+
+	path := utils.PathByUint16LeftPadded(0)
+	payload := utils.LightPayload(11, 12345)
+
+	trie.Insert(path, payload)
+
+	expectedRootHashHex := "b30c99cc3e027a6ff463876c638041b1c55316ed935f1b3699e52a2c3e3eaaab"
```

Comment:
I'd usually put the key, value and expected root hash hex as `const`s at the start of the function in such cases